### PR TITLE
adds missing proxied binaries to the tarball list

### PR DESCRIPTION
**Pull request by amanjeev**
adds proxy symlink to the following:

- rust-gdb
- rust-gdbgui
- rust-lldb
- clippy-driver
- cargo-clippy
- rustfmt
- cargo-fmt
- ferrocene-self-test

To test: 

- Run the full build using `bors try`.
- Then fetch the tarballs from the run's aws s3 bucket
- Untar and check the JSON files in `share/criticaltrust/ferrocene/` 
- The right binaries should have `needs-proxy: true` in the `signed` section/key

Ticket: 86944n9mwhe

### CODE REVIEW

**Comment by amanjeev** [comment]
bors try

**Comment by amanjeev** [comment]
bors try

**Comment by amanjeev** [comment]
bors try

**Comment by amanjeev** [comment]
bors try

**Review by pietroalbini — CHANGES_REQUESTED** [review]
That wasn't supposed to be an approval

File: src/bootstrap/src/core/build_steps/dist.rs [hunk 1]
```diff
@@ -1212,6 +1215,9 @@ impl Step for Clippy {
         tarball.add_file(clippy, "bin", 0o755);
         tarball.add_file(cargoclippy, "bin", 0o755);
         tarball.add_legal_and_readme_to("share/doc/clippy");
+
+        tarball.ferrocene_proxied_binary("bin/clippy");
```
[the remark this answers is not in the document]
Comment (amanjeev) [reply src/bootstrap/src/core/build_steps/dist.rs:1219]: hahaha I assumed those were there for a reason. damn I must not trust the code I get.

**Comment by amanjeev** [comment]
bors try

**Comment by amanjeev** [comment]
This is ready.

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**